defensics/defensics_runner.sh

```diff
@@ -0,0 +1,278 @@
+# Path to suite to run. It should be already installed. Few paths are included in this comment for ease of use
+# SUITE="/opt/Synopsys/Defensics/btle-ad-5.1.0/testtool/btle-ad-510.jar"
+# SUITE="/opt/Synopsys/Defensics/btle-attc-5.3.0/testtool/btle-attc-530.jar"
+# SUITE="/opt/Synopsys/Defensics/btle-att-5.2.0/testtool/btle-att-520.jar"
+# SUITE="/opt/Synopsys/Defensics/btle-hogp-5.1.0/testtool/btle-hogp-510.jar"
+# SUITE="/opt/Synopsys/Defensics/btle-smpc-6.0.0/testtool/btle-smpc-600.jar"
+
+# generate folder name based on chosen suite; can be swapped with any string
+extract_name_helper=${SUITE#*testtool/}
+suite_name=${extract_name_helper%.jar*}
+FOLDER_NAME=$suite_name"/"
+RESULTS_PATH="/home/username/synopsys/defensics/results/"
+# Code for saved results of executed run. In '' user can put special characters that are listed in defensics help
+# (e. g. # for number or letters for date. Not all suites support this formatting
+TEST_CODE="'###'-logs"
+
+# commands to run in instrumentation-external
+BEFORE_RUN="python3 /home/username/BTPTesterCore/defensics/instrumentation_client.py http://localhost:8000/before-run"
+BEFORE_CASE="python3 /home/username/BTPTesterCore/defensics/instrumentation_client.py http://localhost:8000/before-case"
+AS_INSTRUMENTATION="python3 /home/username/BTPTesterCore/defensics/instrumentation_client.py http://localhost:8000/as-instrumentation"
+AFTER_EACH="python3 /home/username/BTPTesterCore/defensics/instrumentation_client.py http://localhost:8000/after-case"
+INSTRUMENT_FAIL="python3 /home/username/BTPTesterCore/defensics/instrumentation_client.py http://localhost:8000/instrument-fail"
+# this has to  be single command with no params
+AFTER_RUN="/home/username/BTPTesterCore/defensics/after-run.py"
+
+# indicies can be in following format:
+# - single integer
+# - test case hash with 0x prefix (e.g. 0x9e9151ba1966f321)
+# - Group name, for example UDP.request
+# - Wildcard * to match multiple groups, e.g. "*.valid" or "UDP.*.element".
+# - Multiple index field entries are separated by commas, e.g. "0,10,20". This can be also generated using get_every_nth_case()
+#   method with arguments: starting index, maximum index, step between test cases
+# This setting overrides SELECTION MODE
+TEST_INDICIES="1-5"
+function get_every_nth_case {
+  local start=$1
+  local end=$2
+  local step=$3
+  local i=1
+  TEST_INDICIES="$start"
+  i=$(( $i+$step ))
+  while [ $i -le $end ]
+  do
+    TEST_INDICIES+=",$i"
+    i=$(( $i+$step ))
+  done
+}
+# if get_every_nth_case not used, comment this line; else, fill with desired values
+#get_every_nth_case 1 2000 10
+if [ "$TEST_INDICIES" != "" ] ; then
+  selection_cmd=""
+  trim_cmd=""
+  test_indicies_cmd="--index $TEST_INDICIES"
+else
+  test_indicies_cmd=""
+fi
+
+# Settings of specific test suites
+# ----------------------------------------------------------------------------------------------------------------------
+# 1. Bluetooth-LE-Advertising-Data
```

Comment:
Maybe those suite specific settings should be in separate functions.

Reply:
This is the same problem as suite names with specific versions, so I think we need to use either seperate functions or variable without version in it.

Reply:
I don't know if there is any profit in putting these in functions, because even if these settings are filled they're only put in final command when SUITE_TYPE variable fits. I don't think that memory usage is a problem here and saving these variables doesn't do much harm.